packages/grid/src/vaadin-grid.js

```diff
@@ -667,12 +638,74 @@ class Grid extends ElementMixin(
     }
 
     this.__intrinsicWidthCache = new Map();
+    // Cache the viewport rows to avoid unnecessary reflows while measuring the column widths
+    const fvi = this._firstVisibleIndex;
```

Comment:
Computing the visible items is an expensive operation because the grid is currently doing it on its own which makes it have to use `getBoundingClientRect`. That being said, the virtualize already provides `firstVisibleIndex` and `lastVisibleIndex` getters. Do you know the reason we aren't using those?

Reply:
IIRC, they don't take the grid header into account properly.

Reply:
Tried running the benchmark tests with `_firstVisibleIndex` and `_lastVisibleIndex` modified to use the virtualizer properties directly, and it seems to have no difference in performance in this case.

Reply:
Compared to the optimized version? I think it sounds more like a refactoring idea, so we can skip it for now.

Reply:
> Compared to the optimized version?

Yes, I get the same performance improvement (~25% on the metric), with this PR regardless of whether `_firstVisibleIndex` uses `getBoundingClientRect()` or `virtualizer.firstVisibleIndex`.